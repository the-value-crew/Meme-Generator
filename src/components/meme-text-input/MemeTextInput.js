import React from 'react'
import "./meme-text-input.scss";
import cog from "../../assets/icons/cog.svg";
import MemeTextSettingsModal from '../meme-text-settings-modal/MemeTextSettingsModal';

class MemeTextInput extends React.Component {
    
    constructor(props){
        super(props);
        this.state = {
            showModal: false
        }
    }
    
    
    onEnter = (e) => {
        if (e.key === 'Enter') {
            this.props.onTextInput(e.target.value)
        }
    };

    showModal = (_) => {
        this.setState(state => { return { showModal: !state.showModal } });

    }

    render() {
        return (
            <div>
                <div className="field">
                    <label className="label" htmlFor={this.props.id}>{this.props.labelText}</label>
                    <div className="control">
                        <input className="input" type="text" id={this.props.id} placeholder={this.props.placeholder} onKeyDown={this.onEnter} />
                        <div className="form-controls">
                            <div className="form-controls__settings" onClick={this.showModal}>
                                <img className="icon mr-0" alt="icon" src={cog}></img>
                            </div>
                        </div>
                    </div>
                </div>
                <MemeTextSettingsModal showModal={this.state.showModal} getMemeSettings={this.getMemeSettings} />
            </div>
        );
    }
}

export default MemeTextInput;
