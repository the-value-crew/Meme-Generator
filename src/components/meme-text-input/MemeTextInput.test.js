import { shallow, render } from 'enzyme';
import React from 'react';
import MemeTextInput from './MemeTextInput';

let shallowWrapper;
let renderWrapper;
beforeEach(() => {
    const onTextInputFunction = jest.fn();
    const mockProps = {
        label: "Text",
        id: "memeText",
        labelText: "Text",
        placeholder: "Text",
        onTextInput: onTextInputFunction
    }
    shallowWrapper = shallow(<MemeTextInput {...mockProps} />);
    renderWrapper = render(<MemeTextInput {...mockProps} />);
})


it('expects to render meme text input', () => {
    expect(renderWrapper).toMatchSnapshot();
});

it('expects to call props on enter key',()=>{
    const onTextInputFunction = jest.fn();
    const mockProps = {
        label: "Text",
        id: "memeText",
        labelText: "Text",
        placeholder: "Text",
        onTextInput: onTextInputFunction
    }
    let shallowWrapper = shallow(<MemeTextInput {...mockProps} />);
    shallowWrapper.find('.input').simulate('keydown', {key: 'Enter', target: {value: "BLAH"}});
    expect(onTextInputFunction).toHaveBeenCalledTimes(1);
    expect(onTextInputFunction).toHaveBeenCalledWith("BLAH");
})

it('expects to not call props on up key',()=>{
    const onTextInputFunction = jest.fn();
    const mockProps = {
        label: "Text",
        id: "memeText",
        labelText: "Text",
        placeholder: "Text",
        onTextInput: onTextInputFunction
    }
    let shallowWrapper = shallow(<MemeTextInput {...mockProps} />);
    shallowWrapper.find('.input').simulate('keydown', {key: 'up', target: {value: "BLAH"}});
    expect(onTextInputFunction).toHaveBeenCalledTimes(0);
})