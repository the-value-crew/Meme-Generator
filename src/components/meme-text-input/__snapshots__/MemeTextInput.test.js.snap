// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render meme text input 1`] = `undefined`;
